controllers/authController.updateProfileController.test.js

```diff
@@ -0,0 +1,234 @@
+import { hashPassword } from "../helpers/authHelper.js";
+import { updateProfileController } from "./authController.js";
+import userModel from "../models/userModel.js";
+
+jest.mock("../models/userModel.js");
+
+const oldProfile = {
+    name: "Halimah Yacob", 
+    email: "yacob@gov.sg", 
+    password: "safePassword", 
+    address: "Istana, Orchard Road, Singapore 238823", 
+    phone: "999"
+};
+
+const oldProfileUpdate = {
+    name: "Halimah Yacob", 
+    password: "safePassword", 
+    address: "Istana, Orchard Road, Singapore 238823", 
+    phone: "999"
+}
+
+const sampleName = "John Doe";
+const sampleEmail = "johndoe@gmail.com";
+const passwordLenMoreThan6 = "password";
+const passwordLen6 = "passwo";
+const passwordLen5 = "passw";
+const passwordHash = "hashedPassword";
+const validAddress = "123 Main St, Springfield, IL 62701";
+const validPhone = "99999999";
+const invalidPhone = "abc";
+
+const newProfile = {
+    name: sampleName, 
+    email: sampleEmail, 
+    password: passwordLenMoreThan6, 
+    address: validAddress, 
+    phone: validPhone
+}
+
+const newProfileUpdate = {
+    name: sampleName, 
+    password: passwordHash, 
+    address: validAddress, 
+    phone: validPhone
+}
+
+const req = {
+    body: {}, 
+    user: { _id: "123" }
+}
+
+const res = {
+    json: jest.fn(),
+    status: jest.fn(() => res),
+    send: jest.fn()
+};
+
+jest.mock("./../helpers/authHelper.js", () => ({
+    hashPassword: jest.fn()
+}));
+
+
+// Tests for updateProfileController
+describe("updateProfileController", () => {
+    beforeAll(() => {
+        hashPassword.mockReturnValue(passwordHash);
+        userModel.findById.mockReturnValue(oldProfile);
+        userModel.findByIdAndUpdate.mockImplementation(() => jest.fn());
+    })
+
+    beforeEach(() => {
+        jest.clearAllMocks();
+        req.body = JSON.parse(JSON.stringify(newProfile));
+    });
+
+    it("should update user's profile", async () => {
+        await updateProfileController(req, res);
+        expect(userModel.findByIdAndUpdate).toHaveBeenCalledTimes(1);
+        expect(userModel.findByIdAndUpdate).toHaveBeenCalledWith(req.user._id, newProfileUpdate, { new: true });
+        expect(res.status).toHaveBeenCalledWith(200);
+        expect(res.send).toHaveBeenCalledTimes(1);
+    });
+
+    it("should update user's profile with new password of length 6", async () => {
+        req.body.password = passwordLen6;
+
+        let updatedUser = JSON.parse(JSON.stringify(newProfileUpdate));
+        updatedUser.password = passwordHash;
+
+        await updateProfileController(req, res);
+
+        expect(userModel.findByIdAndUpdate).toHaveBeenCalledTimes(1);
+        expect(userModel.findByIdAndUpdate).toHaveBeenCalledWith(req.user._id, updatedUser, { new: true });
+        expect(res.status).toHaveBeenCalledWith(200);
+        expect(res.send).toHaveBeenCalledTimes(1);
```

Comment:
Not sure if you want to assert the content of the response, such as `success` and `message` attributes.

Reply:
Will that make the test too brittle?